spec.bs

```diff
@@ -4372,319 +4380,296 @@ value is used to denote that the currency is unspecified.
+    {{AuctionAdConfig/additionalBids}} {{Promise}} is [=rejected=].
 </dl>
 
 <div algorithm>
-To <dfn>wait until configuration input promises resolve</dfn> given an [=auction config=] |auctionConfig|:
-1. Wait until |auctionConfig|'s [=auction config/pending promise count=] is 0.
-1. [=Assert=] |auctionConfig|'s [=auction config/auction signals=], [=auction config/seller signals=],
-  [=auction config/per buyer signals=], [=auction config/per buyer currencies=],
-  [=auction config/per buyer timeouts=], [=auction config/per buyer cumulative timeouts=], and
-  [=auction config/direct from seller signals header ad slot=] are not {{Promise}}s, and
-  [=auction config/expects additional bids=] is false.
-1. If |auctionConfig|'s [=auction config/auction signals=], [=auction config/seller signals=],
-  [=auction config/per buyer signals=], [=auction config/per buyer currencies=],
-  [=auction config/per buyer timeouts=], [=auction config/per buyer cumulative timeouts=], or
-  [=auction config/direct from seller signals header ad slot=] is failure, return failure.
-1. Return.
+To <dfn>wait until configuration input promises resolve</dfn> given an [=auction config=]
+|auctionConfig|:
 
+  1. Wait until |auctionConfig|'s [=auction config/pending promise count=] is 0.
+  1. [=Assert=] |auctionConfig|'s [=auction config/auction signals=], [=auction config/seller signals=],
+    [=auction config/per buyer signals=], [=auction config/per buyer currencies=],
+    [=auction config/per buyer timeouts=], [=auction config/per buyer cumulative timeouts=], and
+    [=auction config/direct from seller signals header ad slot=] are not {{Promise}}s, and
+    [=auction config/expects additional bids=] is false.
+  1. If |auctionConfig|'s [=auction config/auction signals=], [=auction config/seller signals=],
+    [=auction config/per buyer signals=], [=auction config/per buyer currencies=],
+    [=auction config/per buyer timeouts=], [=auction config/per buyer cumulative timeouts=], or
+    [=auction config/direct from seller signals header ad slot=] is failure, return failure.
+  1. Return.
 </div>
 
 <div algorithm>
 To <dfn>recursively wait until configuration input promises resolve</dfn> given an [=auction config=]
 |auctionConfig|:
-1. [=list/For each=] |componentAuctionConfig| in |auctionConfig|'s [=auction config/component auctions=]:
-  1. If the result of [=waiting until configuration input promises resolve=] given |componentAuctionConfig| is
-    failure, return failure.
-1. Return the result of [=waiting until configuration input promises resolve=] given |auctionConfig|.
 
+  1. [=list/For each=] |componentAuctionConfig| in |auctionConfig|'s
+    [=auction config/component auctions=]:
+    1. If the result of [=waiting until configuration input promises resolve=] given
+      |componentAuctionConfig| is failure, return failure.
+  1. Return the result of [=waiting until configuration input promises resolve=] given |auctionConfig|.

```

Comment:
Do you mean to "wait until configuration input promises resolve" many times here? We wait many times in the above loop, and once here. Is that a problem / should we just wait "once" for all of them?

Reply:
inside the loop, it waits for |componentAuctionConfig|'s promises. Here outside the loop, it waits for |auctionConfig|'s promises.

Reply:
ACK, thanks.